contracts/Certificate.sol

```diff
@@ -335,7 +345,9 @@ contract Certificate is
       certificateId: certificateId,
       certificateAmount: certificateAmount,
       removalIds: removalIds,
-      removalAmounts: removalAmounts
+      removalAmounts: removalAmounts,
+      purchasingTokenAddress: purchasingTokenAddress,
```

Comment:
I'd also be curious how much additional gas overhead was added due to the number of arguments and the non-tighty packed nature of these arguments

Reply:
parameter ordering doesn't matter in terms of packing and gas: [Here](https://docs.soliditylang.org/en/v0.8.17/internals/layout_in_storage.html#storage-inplace-encoding) it says "When dealing with function arguments or memory values, there is no inherent benefit because the compiler does not pack these values."